packages/components/src/sortTabs/SortTab.js

```diff
@@ -0,0 +1,174 @@
+// @flow
+
+import * as React from 'react';
+import { View } from 'react-native';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import {
+  Text,
+  StyleSheet,
+  Icon,
+  Hoverable,
+  designTokens,
+} from '@kiwicom/universal-components';
+
+import TouchableWithoutFeedback from '../TouchableWithoutFeedback';
+import type { SortIconTypes, SortTypes, PriceDurationInfo } from './SortTabs';
+
+export type Props = {|
+  +label: string,
+  +value: SortTypes,
+  +isFirst: boolean,
+  +isLast: boolean,
+  +isActive: boolean,
+  +priceDurationInfo: ?PriceDurationInfo,
+  +icon: SortIconTypes,
+  +onPress: SortTypes => void,
+|};
+
+type State = {|
+  isHovered: boolean,
+|};
+
+export default class SortTab extends React.Component<Props, State> {
+  state = {
+    isHovered: false,
+  };
+
+  handleOnMouseEnter = () => {
+    this.setState({ isHovered: true });
+  };
+
+  handleOnMouseLeave = () => {
+    this.setState({ isHovered: false });
+  };
+
+  handleOnPress = () => {
+    const { value, onPress } = this.props;
+    onPress(value);
+  };
+
+  render() {
+    const {
+      label,
+      isFirst,
+      isLast,
+      isActive,
+      icon,
+      priceDurationInfo,
+    } = this.props;
+    const { isHovered } = this.state;
+    return (
+      <Hoverable
+        onMouseEnter={this.handleOnMouseEnter}
+        onMouseLeave={this.handleOnMouseLeave}
+      >
+        <TouchableWithoutFeedback onPress={this.handleOnPress}>
+          <View
+            style={[
+              styles.container,
+              isFirst && styles.containerFirst,
+              isLast && styles.containerLast,
+              isActive && styles.containerActive,
+              isHovered && styles.containerHovered,
+            ]}
+          >
+            <View>
+              <Text
+                style={[styles.label, isActive && styles.labelActive]}
+                align="left"
+                numberOfLines={1}
+              >
+                {label}
+              </Text>
+              <View style={styles.priceDurationContainer}>
+                {priceDurationInfo?.price || priceDurationInfo?.currency ? (
```

Comment:
Thinking if that will be more readable and have better performance with deconstruction?
`const {currency, price, duration} = priceDurationInfo`

Reply:
Yes I thought of that but with them being potentially null, Flow complains 👎

Reply:
🤔 right
so maby with this it will pass
```

    const price = priceDurationInfo?.price;
    const currency = priceDurationInfo?.currency;
    const duration = priceDurationInfo?.duration
```

Reply:
Changed!